debezium-core/src/main/java/io/debezium/pipeline/EventDispatcher.java

```diff
@@ -118,7 +126,15 @@ public void dispatchDataChangeEvent(T dataCollectionId, ChangeRecordEmitter chan
                 throw new IllegalArgumentException("No metadata registered for captured table " + dataCollectionId);
             }
 
-            changeRecordEmitter.emitChangeRecords(dataCollectionSchema, streamingReceiver);
+            changeRecordEmitter.emitChangeRecords(dataCollectionSchema, new Receiver() {
+
+                @Override
+                public void changeRecord(DataCollectionSchema schema, Operation operation, Object key, Struct value,
+                        OffsetContext offset) throws InterruptedException {
+                    eventListener.onEvent("operation = " + operation + ", source = " + dataCollectionId + ", id = " + key + ", offset = " + offset);
```

Comment:
Ok, so that slightly changes the notion of "event", whereas it used to be at the level of the original event we received, it's not at the level of emitted events. There might be a difference in case of PK updates and deletions (tombstone events). Fine for me, in case it was a conscious decision on your end? It's a bit inconsistent with MySQL for the time being, but hopefully we'll have migrated that connector to the framework, too.

Reply:
@gunnarmorling Hi, yes you are right in this case - but if you look at https://github.com/shyiko/mysql-binlog-connector-java/blob/master/src/main/java/com/github/shyiko/mysql/binlog/jmx/BinaryLogClientStatistics.java#L49 then we might provide similar level of information. I intentionally moved the semantics a little bit as for example in case of POstgres there is no queivalent of type in the source event. In case of SQL Server there is a more or less 1:1 mapping between source and logical type.
So I propose in mid-term unite all connectors on logical type and introduce a new metric (where it makes sense) that would expose the native event if there will be a demand for it.